app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java

```diff
@@ -91,24 +91,31 @@ private PlayerHelper() {
 
     ////////////////////////////////////////////////////////////////////////////
     // Exposed helpers
-    ////////////////////////////////////////////////////////////////////////////
+
+    /// /////////////////////////////////////////////////////////////////////////
 
     @NonNull
     public static String getTimeString(final int milliSeconds) {
-        final int seconds = (milliSeconds % 60000) / 1000;
-        final int minutes = (milliSeconds % 3600000) / 60000;
-        final int hours = (milliSeconds % 86400000) / 3600000;
-        final int days = (milliSeconds % (86400000 * 7)) / 86400000;
+        final int SECONDS_IN_MINUTE = 60;
+        final int SECONDS_IN_HOUR = 3600;
+        final int SECONDS_IN_DAY = 86400;
+
+        int totalSeconds = milliSeconds / 1000;
 
-        STRING_BUILDER.setLength(0);
-        return (days > 0
-                ? STRING_FORMATTER.format("%d:%02d:%02d:%02d", days, hours, minutes, seconds)
-                : hours > 0
-                ? STRING_FORMATTER.format("%d:%02d:%02d", hours, minutes, seconds)
-                : STRING_FORMATTER.format("%02d:%02d", minutes, seconds)
-        ).toString();
+        int days = totalSeconds / (SECONDS_IN_DAY);
+        int hours = (totalSeconds % (SECONDS_IN_DAY)) / SECONDS_IN_HOUR;
+        int minutes = (totalSeconds % SECONDS_IN_HOUR) / SECONDS_IN_MINUTE;
+        int seconds = totalSeconds % SECONDS_IN_MINUTE;
+
+        if (days > 0) {
+            return STRING_FORMATTER.format("%d:%02d:%02d:%02d", days, hours, minutes, seconds).toString();
+        } else if (hours > 0) {
+            return STRING_FORMATTER.format("%d:%02d:%02d", hours, minutes, seconds).toString();
+        }
+        return STRING_FORMATTER.format("%02d:%02d", minutes, seconds).toString();
     }
```

Comment:
This can be replaced with [`DateUtils.formatElapsedTime(long)`](https://developer.android.com/reference/android/text/format/DateUtils#formatElapsedTime(long)).